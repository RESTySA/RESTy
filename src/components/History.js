import React from 'react';
import PropTypes from 'prop-types';

const History = ({ method, url }) => {
  return (
    <li>
      <h2>{method.toUpperCase()}</h2>
      <span>{url}</span>
    </li>
  );
};

History.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

export default History;