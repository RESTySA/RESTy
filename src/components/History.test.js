import React from 'react';
import { shallow } from 'enzyme';
import History from './History';

describe('History component', () => {
  const test = {
    url: 'test.com',
    method: 'get'
  };

  it('renders History', () => {
    const wrapper = shallow(<History url={test.url} method={test.method} />);
    expect(wrapper).toMatchSnapshot();
  });
});