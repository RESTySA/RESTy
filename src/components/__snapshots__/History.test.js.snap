// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<Fragment>
  <h4>
    GET
  </h4>
  <span>
    test.com
  </span>
</Fragment>
`;
